### PR TITLE
[stop:1] Add build stop on done and refresh SCM repo on pipeline save

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Couple changes:
- `build.stop` just stops the job now (doesn't update the build or github status)
- `build.update` will stop the job if the build is done (helps cleanup old resources)
- `pipeline.sync` will update the `scmRepo` object

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -214,38 +214,43 @@ class BuildModel extends BaseModel {
...
+                .then(() => this.pipeline)
+                .then(pipeline => this.updateCommitStatus(pipeline));
+        }
 
-                // update scm with status
-                return this.pipeline
-                    .then(pipeline => this.updateCommitStatus(pipeline))
-                    .then(() => this);
-            });
+        return prom
```
Comment (d2lam) [inline lib/build.js:232]: You need `prom` so that it only returns after the above promises resolve?

File: lib/build.js [hunk 2]
```diff
@@ -214,38 +214,43 @@ class BuildModel extends BaseModel {
...
+            .then(() => super.update())
+            .then(() => this);
     }
 
     /**
-     * Stop a build and update github status as failure
+     * Stop a build
```
Comment (d2lam) [inline lib/build.js:238]: What is the reason for not updating github status anymore?

File: lib/build.js [hunk 3]
```diff
@@ -214,38 +214,43 @@ class BuildModel extends BaseModel {
...
-
-        // stop the build
         return this[executor].stop({ buildId: this.id })
-            // update status
-            .then(() => this.update());
+            .then(() => this);
+    }
+
+    /**
+     * Check if a build is doesntMatter
```
Comment (d2lam) [inline lib/build.js]: doesntMatter?

**Review by d2lam — CHANGES_REQUESTED** [review]

File: lib/build.js (hunk 2 above)
Comment (stjohnjohnson) [reply lib/build.js:238]: So that stop just stops the build, update is the funnel for updating github status.

File: lib/build.js (hunk 1 above)
Comment (stjohnjohnson) [reply lib/build.js:232]: I need it so I can conditionally chain events.

File: lib/pipeline.js [hunk 4]
```diff
@@ -272,6 +274,34 @@ class PipelineModel extends BaseModel {
...
     }
+
+    /**
+     * Update the SCM Repo based on the SCM URL
+     * @method refreshScmRepo
+     * @return {Promise}
+     */
+    refreshScmRepo() {
+        // Skip if no changes
+        if (!this.isDirty('scmUrl')) {
```
Comment (d2lam) [inline lib/pipeline.js]: Wouldn't this skip the very first time?

**Review by d2lam — CHANGES_REQUESTED** [review]

**Review by d2lam — APPROVED** [review]
👍  LGTM

**Pull request merged**